### ISSUE TITLE
Custom type with underlying type returns error

### ISSUE THREAD

**Issue by mkasner** [go-goracle/goracle#96]
When type of arg is not custom struct but just CustomType with underlying builtin type goracle returns error.
Example test
```go
// Underlying is an example of custom type with underlying builtin type
type Underlying int64

func TestSelectUnderlyingType(t *testing.T) {
	t.Parallel()
	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
	defer cancel()
	conn, err := testDb.Conn(ctx)
	if err != nil {
		t.Fatal(err)
	}
	defer conn.Close()
	const num = 10
	var nums Underlying = 10
	rows, err := testDb.QueryContext(ctx, "SELECT object_name, object_type, object_id, created FROM user_objects WHERE ROWNUM <= NVL(:alpha, 2) ORDER BY object_id", sql.Named("alpha", nums))
	if err != nil {
		t.Fatalf("%+v", err)
	}
	n, oldOid := 0, int64(0)
	for rows.Next() {
		var tbl, typ string
		var oid int64
		var created time.Time
		if err := rows.Scan(&tbl, &typ, &oid, &created); err != nil {
			t.Fatal(err)
		}
		t.Log(tbl, typ, oid, created)
		if tbl == "" {
			t.Fatal("empty tbl")
		}
		n++
		if oldOid > oid {
			t.Errorf("got oid=%d, wanted sth < %d.", oid, oldOid)
		}
		oldOid = oid
	}
	if n == 0 || n > num {
		t.Errorf("got %d rows, wanted %d", n, num)
	}
}
```
Outputs error
```
z_test.go:1743: 1. arg: unknown type goracle_test.Underlying
```
rana/ora.v4 works with this example

related to #94

**Comment by mkasner** [comment]
So simple custom types don't have to be converted to their underlying types explicitly.

It is usual behavior with other drivers.

**Comment by tgulacsi** [comment]
How do they do it?
It's just magic: the copying has to be done, and this way you shovel it into the driver...

**Comment by mkasner** [comment]
I didn't inspect how they do it.
You have a point with 'magic'. I tried to implement it and it's not possible without reflect and observing Kind.
If you don't want that approach in this driver, it's fine with me. I'll convert values or provide Valuer interface for those types.

**Comment by tgulacsi** [comment]
Can you try 9a6ece4 ?

**Comment by mkasner** [comment]
It works. So you decided to support magic?

**Comment by tgulacsi** [comment]
I've released it as v2.6.0, but hidden this magic under the "MagicTypeConversion()" option.

**Comment by mkasner** [comment]
thanks for your quick response on this
